crates/blockchain/payload.rs

```diff
@@ -228,21 +252,58 @@ pub fn build_payload(
     debug!("Building payload");
     let mut evm_state = evm_state(store.clone(), payload.header.parent_hash);
     let mut context = PayloadBuildContext::new(payload, &mut evm_state)?;
-    apply_withdrawals(&mut context)?;
-    fill_transactions(&mut context)?;
-    finalize_payload(&mut context)?;
+    let mut block_cache = apply_withdrawals(&mut context)?;
+    fill_transactions(&mut context, &mut block_cache)?;
+    finalize_payload(&mut context, &mut block_cache)?;
     Ok((context.blobs_bundle, context.block_value))
 }
 
-pub fn apply_withdrawals(context: &mut PayloadBuildContext) -> Result<(), EvmError> {
-    // Apply withdrawals & call beacon root contract, and obtain the new state root
-    let spec_id = spec_id(&context.chain_config()?, context.payload.header.timestamp);
-    if context.payload.header.parent_beacon_block_root.is_some() && spec_id == SpecId::CANCUN {
-        beacon_root_contract_call(context.evm_state, &context.payload.header, spec_id)?;
+pub fn apply_withdrawals(
+    context: &mut PayloadBuildContext,
+) -> Result<HashMap<Address, Account>, EvmError> {
```

Comment:
How about
```suggestion
) -> Result<CacheDB, EvmError> {
```

This would need the following suggestion as well I believe.

Do you like it better or do you prefer to leave the Hash as is?